R/scoring.R

```diff
@@ -68,77 +62,29 @@ evaluate_predictions <- function(
   assert_that("actual" %in% names(truth_data),
     msg = paste("`truth_data` must contain a column named `actual`")
   )
-  predictions_cards <- left_join(predictions_cards, truth_data,
-    by = intersect(colnames(predictions_cards), colnames(truth_data))
-  )
-  if (is.null(err_measures) || length(err_measures) == 0) {
-    score_card <- predictions_cards
-  } else {
-    err_calls <- lapply(err_measures, function(err_measure) {
-      quo(err_measure(.data$quantile, .data$value, .data$actual))
-    })
-    score_card <- predictions_cards %>%
-      group_by(across(all_of(grp_vars))) %>%
-      summarize(!!!err_calls, .groups = "drop") %>%
-      inner_join(predictions_cards, by = grp_vars, multiple = "all")
-  }
+
+  grp_vars <- c("geo_value", "forecast_date", "target_end_date")
+  err_calls <- lapply(err_measures, function(err_measure) {
+    quo(err_measure(.data$quantile, .data$value, .data$actual))
+  })
+  predictions_cards <- left_join(predictions_cards, truth_data)
```

Comment:
this file could use more comments, and I don't think it has any tests? not sure I follow the changes. I guess for this line in particular, the `by = intersect(colnames(predictions_cards), colnames(truth_data)` was redundant.

Reply:
yea, it was all copy pasta from evalcast, but i tested it all in my repl as I was changing it. there were a few lines that were redundant (like this one) but most of the rest of the changes are about centralizing the work in one readable function, rather than many.

Reply:
if it's straightforward to add the repl tests as tests I think that would be a good idea. Maybe copy some tests from evalcast (assuming evalcast has tests)

Reply:
moved over to scoringutils (compared that their scores of an already done forecast match already computed scores for the same forecast numerically), so no need to test.